tests/test_multidict.py

```diff
@@ -387,6 +387,17 @@ def test_and2(self, cls: Type[MutableMultiMapping[str]]) -> None:
 
         assert {"key"} == {"key", "key2"} & d.keys()
 
+    def test_and_not_implemented(self, cls: Type[MutableMultiMapping[str]]) -> None:
+        d = cls([("key", "value1")])
+
+        with pytest.raises(TypeError, match=r"unsupported operand type(\(s\))? for \&"):
+            d.keys() & 1
```

Comment:
this test case is just to cover the `NotImplemented` that shows up as `TypeError`, with your sugestions we dont cover that line, do we?

```python
if not isinstance(other, Iterable):
        return NotImplemented
```

Reply:
@a5r0n yeah, but your previous implementation didn't either, right? Anyway, you can't cover 2 `return` statements with a single invocation. So that would need to be addressed separately. Are you able to measure coverage locally, to see which lines are actually hit?

Reply:
i didnt understand you eralier, your suggestion works fine.

the last line that dont have coverage is `if isinstance(view, Set): `